pkgs/development/libraries/gfal2/default.nix

```diff
@@ -0,0 +1,92 @@
+{ lib
+, stdenv
+, callPackage
+, fetchFromGitHub
+  # Native build inputs
+, cmake
+, pkg-config
+  # General build inputs
+, glib
+, gtest
+, json_c
+, openldap
+  # Plugin build inputs
+, davix-copy
+, dcap
+, libssh2
+, libuuid
+, xrootd
+  # Flags
+, doCheck ? true
+, ...
```

Comment:
Use of elipsis is discouraged for functions used in `callPackage`.

Reply:
It could also be `pluginEnableStatus` attrset. But why is it not encouraged? IIUC, `callPackage` would only supply the arguments explicitly specified when defining the function.

Reply:
I think we want to let nix check arguments validity. If you make a typo in the name, the nix will spew `called with unexpected argument` error at eval time, but only if ellipsis is not not included.

Reply:
Addressed by rewriting with fixed-point arguments (the `finalAttrs: { }` style of input arguments).